metadata-ingestion/src/datahub/ingestion/glossary/classification_mixin.py

```diff
@@ -289,12 +291,22 @@ def classification_workunit_processor(
     table_id: List[str],
     data_reader_kwargs: dict = {},
 ) -> Iterable[MetadataWorkUnit]:
+    """
+    Classification handling for a particular table.
+    Currently works only for workunits having MCP or MCPW
+    """
     table_name = ".".join(table_id)
     if not classification_handler.is_classification_enabled_for_table(table_name):
         yield from table_wu_generator
     for wu in table_wu_generator:
         maybe_schema_metadata = wu.get_aspect_of_type(SchemaMetadata)
-        if maybe_schema_metadata:
+        if (
+            isinstance(wu.metadata, MetadataChangeEvent)
+            and len(wu.metadata.proposedSnapshot.aspects) > 1
+        ) or not maybe_schema_metadata:
+            yield wu
```

Comment:
shouldn't we remove the SchemaMetadata aspect from the snapshot, since we're emitting it after

Reply:
For now, this is leaving the MCEs as is and not processing it.

Reply:
In followup, I am planning to add support for classifying SchemaMetadata in MCE and refractoring sql_common to use this.